x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/job_messages_pane.tsx

```diff
@@ -77,13 +81,13 @@ export const JobMessagesPane: FC<JobMessagesPaneProps> = React.memo(
           })
         );
       }
-    }, [jobId]);
+    }, [jobId, notificationIndices.length]);
```

Comment:
Is it safe to rely on the array length here? Shall we check if the content of the `notificationIndices` has been changed instead?